contracts/allocators/alchemistAllocator.sol

```diff
@@ -0,0 +1,210 @@
+
+    function getStakeTotalUnclaimed(address _account, uint256 _poolId) external view returns (uint256);
+
+    function deposit(uint256 _poolId, uint256 _depositAmount) external;
+
+    function withdraw(uint256 _poolId, uint256 _withdrawAmount) external;
+}
+
+/**
+ *  Contract deploys Alchemist from treasury into the Tokemak tALCX pool,
+ *  tALCX contract gives tALCX token in ratio 1:1 of Alchemist token deposited,
+ *  Contract stake tALCX token on Alchemist staking pool and earn ALCX as reward,
+ *  Contract claims reward and compound it,
+ *  Contract withdraw funds from Alchemist staking pool and Tokemak tALCX pool,
+ *  Sends back Alchemist token with accured reward to treasury.
+ */
+
+contract AlchemistAllocator is Ownable {
+    /* ======== DEPENDENCIES ======== */
+
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /* ======== STATE VARIABLES ======== */
+
+    address alchemist;
+    address tokemak_manager = 0xA86e412109f77c45a3BC1c5870b880492Fb86A14;
+
+    ITokemak_tALCX immutable tALCX; // Tokemak tALCX deposit contract
+    IStakingPools immutable pool; // Alchemist staking contract
+    ITreasury immutable treasury; // Olympus Treasury
+
+    uint256 public totalAlchemistDeposited;
+
+    /* ======== CONSTRUCTOR ======== */
+
+    constructor(
+        address _treasury,
+        address _alchemist,
+        address _tALCX,
+        address _pool
+    ) {
+        require(_treasury != address(0));
+        treasury = ITreasury(_treasury);
+
+        require(_tALCX != address(0));
+        tALCX = ITokemak_tALCX(_tALCX);
+
+        require(_pool != address(0));
+        pool = IStakingPools(_pool);
+
+        alchemist = _alchemist;
+    }
+
+    /* ======== POLICY FUNCTIONS ======== */
+
+    /**
+     *  @notice sets Alchemist address
+     */
+    function setAlchemistAddress(address _alchemist) external onlyOwner {
+        alchemist = _alchemist;
+    }
+
+    /**
+     *  @notice compound reward by claiming pending rewards and
+     *      calling the deposit function
+     *  @param _poolId pool id of tALCX on Alchemist staking pool
+     */
+    function compoundReward(uint256 _poolId) public onlyOwner {
+        pool.claim(_poolId);
+        uint256 alchemist_balance = IERC20(alchemist).balanceOf(address(this));
+
+        require(alchemist_balance > 0, "contract has no alchemist token");
+        deposit(alchemist_balance, _poolId, true);
+    }
+
+    /**
+     *  @notice withdraws asset from treasury, deposits asset into Tokemak tALCX,
+     *      then deposits tALCX into Alchemist staking pool
+     *  @param amount amount to deposit
+     *  @param _poolId pool id of tALCX on Alchemist staking pool
+     *  @param isCompounding used to indicate if the contract is compounding pending rewards
+     */
+    function deposit(
+        uint256 amount,
+        uint256 _poolId,
+        bool isCompounding
+    ) public onlyOwner {
+        if (isCompounding) {} else {
```

Comment:
Can remove else statement by checking for `!isCompounding`

Reply:
Adopted